### ISSUE TITLE
npm install gives error

### ISSUE THREAD

**Issue by Owumaro** [tfeng/collection#3]
Hey, I'm trying to use collection but can't install it.
I'm using node v0.10.32, npm v1.4.28 (latests) on OS X 10.9.5. I tryed to update gcc/g++ (v4.9.1) but nothing changed.
Here is the log when I try `npm install collection`.

```
> collection@0.1.5 install /Users/xxx/node_modules/collection
> node-gyp rebuild

  CXX(target) Release/obj.target/NativeTypes/src/common.o
../src/common.cc: In instantiation of 'v8::Handle<v8::Value> Collection<Storage>::Iterate(v8::Handle<v8::Value> (*)(const v8::Arguments&), const v8::Arguments&) [with Storage = std::map<v8::Persistent<v8::Value>, v8::Persistent<v8::Value>, ValueComparator>]':
../src/common.cc:806:16:   required from here
../src/common.cc:205:5: error: exception handling disabled, use -fexceptions to enable
   } catch (...) {
     ^
make: *** [Release/obj.target/NativeTypes/src/common.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:810:12)
gyp ERR! System Darwin 13.4.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/xxx/node_modules/collection
gyp ERR! node -v v0.10.32
gyp ERR! node-gyp -v v1.0.1
gyp ERR! not ok 

npm ERR! collection@0.1.5 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the collection@0.1.5 install script.
npm ERR! This is most likely a problem with the collection package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls collection
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 13.4.0
npm ERR! command "node" "/usr/local/bin/npm" "install" "collection"
npm ERR! cwd /Users/xxx
npm ERR! node -v v0.10.32
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! not ok code 0
```

Thanks

**Comment by tfeng** [comment]
Hmm... It works for me. I'm using OSX 10.9.5 too.

``` bash
$ npm install collection
|
> collection@0.1.5 install /Users/tfeng/tmp/test/node_modules/collection
> node-gyp rebuild

  CXX(target) Release/obj.target/NativeTypes/src/common.o
  CXX(target) Release/obj.target/NativeTypes/src/NativeTypes.o
  CXX(target) Release/obj.target/NativeTypes/src/Map.o
  CXX(target) Release/obj.target/NativeTypes/src/Set.o
  CXX(target) Release/obj.target/NativeTypes/src/Vector.o
  SOLINK_MODULE(target) Release/NativeTypes.node
  SOLINK_MODULE(target) Release/NativeTypes.node: Finished
collection@0.1.5 node_modules/collection
└── bindings@1.2.1
tfeng:~/tmp/test $ node -v
v0.10.32
tfeng:~/tmp/test $ npm -v
1.4.28
tfeng:~/tmp/test $ gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.51) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
```

Did you install node.js using the native package at http://nodejs.org/download/, or did you install using macport or brew?

**Comment by tfeng** [comment]
Can you try this and see whether it fixes the issue for you? Whether or not I do this, the code compiles in my environment; but it may fix your problem. Let me know.
1. Check out the git repo https://github.com/tfeng/collection.git
2. Apply this patch:

``` patch
diff --git binding.gyp binding.gyp
index 943a6e2..55d6c55 100644
--- binding.gyp
+++ binding.gyp
@@ -11,6 +11,13 @@
         ['OS=="linux"', {
             'cflags_cc!': ['-fno-exceptions']
         }],
+        ['OS=="mac"', {
+          'xcode_settings': {
+            'OTHER_CFLAGS': [
+              '-fexceptions'
+            ]
+          }
+        }],
         ['OS=="win"', {
           'msvs_settings': {
             'VCCLCompilerTool': {
```

After this, run `npm install` in the root directory to see whether it passes.

**Comment by Owumaro** [comment]
With your patch, I get this :

```
> collection@0.1.5 install /Users/xxx/collection
> node-gyp rebuild

  CXX(target) Release/obj.target/NativeTypes/src/common.o
  CXX(target) Release/obj.target/NativeTypes/src/NativeTypes.o
  CXX(target) Release/obj.target/NativeTypes/src/Map.o
  CXX(target) Release/obj.target/NativeTypes/src/Set.o
  CXX(target) Release/obj.target/NativeTypes/src/Vector.o
  SOLINK_MODULE(target) Release/NativeTypes.node
  SOLINK_MODULE(target) Release/NativeTypes.node: Finished
```

Looks good. However if I run node after that (in the same folder), it cannot find the module (not sure if I'm doing it right).

```
Error: Cannot find module 'collection'
```

**Comment by tfeng** [comment]
That is fine. It means the code compiles with this flag. Can you confirm that, without this patch, the compilation fails when you run npm install in the root directory?

If so, I can make this change. On my macbook, it works with or without the patch.

**Comment by Owumaro** [comment]
Yes, without the patch, `npm install` at the root of collection gives the same error as before.

**Comment by tfeng** [comment]
Released 0.1.6 to fix this issue.
